server/sttp-stub-server/src/main/scala/sttp/tapir/server/stub/package.scala

```diff
@@ -0,0 +1,28 @@
+package sttp.tapir.server
+
+import sttp.client.monad.MonadError
+import sttp.client.monad.syntax._
+import sttp.client.{NothingT, SttpBackend, _}
+import sttp.tapir.client.sttp._
+import sttp.tapir.server.internal.{DecodeInputs, DecodeInputsResult}
+
+package object stub extends SttpStubServer {
+
+  implicit class StubbedEndpoint[I, E, O, F[_]: MonadError](
+      serverEndpoint: ServerEndpoint[I, E, O, Nothing, F]
+  ) {
+    def testUsing(input: I): Option[F[Either[E, O]]] = {
+      implicit val stubBackend: SttpBackend[F, Nothing, NothingT] = List(serverEndpoint).toBackendStub
+      val req = serverEndpoint.endpoint.toSttpRequestUnsafe(uri"http://test.com").apply(input)
+
+      val decodeInputResult = DecodeInputs(serverEndpoint.endpoint.input, new SttpDecodeInput(req))
+      decodeInputResult match {
+        case DecodeInputsResult.Failure(_, _) =>
+          None
+        case DecodeInputsResult.Values(_, _) =>
+          Some(req.send().map(_.body))
+      }
```

Comment:
I'm wondering what exactly we are testing where :)
So far I've got (mainly a sanity check):
1. `endpoint.toSttpRequestUnsafe.apply` tests input encoding from high-level to the "raw" format
2. `DecodeInputs(serverEndpoint.endpoint.input, new SttpDecodeInput(req))` tests input decoding from raw to high-level formats

We don't test output encoding/decoding as the values are passed straight from the server logic to sttp's response, right? Maybe we should? ;)

Reply:
We could test this, but how we should react to encoding/decoding errors? Also with `None`? Or maybe we should fail `F`?

Reply:
Well I started wondering what's exactly the utility of `endpoint.testUsing` ;) Is it just invoking the server logic function? That would be quite pointeless ;). 

So I was having second thoughts about the whole `.testUsing`. But maybe testing encoding & decoding is a good use-case? But then again, why go through sttp client etc - couldn't we just encode & decode all inputs, call the server login, and encode & decode the outputs?

As for possible error reporting - I think a failed F would be appropriate, as that's what you'll get when an input fails in encoding/decoding?

(the situation with `SttpStubServer` is quite different, as it will be very useful for testing services which call other APIs using sttp client)

Reply:
Good point about whare we are actually testing ;) Server logic can be easily extracted from a `ServerEndpoint` and tested in isolation. And adding input/output encoding/decoding can also be performed around without using stub backends etc. This may simplify the implementation.

One more case that I can think of is testing a list of endpoints converted to a single "stub backend", but has anyone ever requested such a feature? :)

Reply:
Maybe this should be another way around, like:
```
implicit val testingBackend = SttpBackendStub.synchronous
  .whenRequestMatches(someEndpoint)
  .thenRespond("Hello there!")
  .whenRequestMatches(otherEndpoint)
  .thenRespondServerError()
```
?